File: models/gameActions.js
// Imports
const db = require('./db');

exports.create = (data, done) => {
    const sql = 'INSERT INTO game_actions (game_id, action) VALUES (?, ?)';
    const values = [data.gameID, data.action];
    db.query(sql, values, (error, results, fields) => {
        if (error) {
            done(error, data);
            return;
        }

        done(null, data);
    });
};

exports.getAll = (socket, data, done) => {
    const sql = `
        SELECT action
        FROM game_actions
        WHERE game_id = ?
        ORDER BY id
    `;
    const values = [data.gameID];
    db.query(sql, values, (error, results, fields) => {
        if (error) {
            done(error, socket, data);
            return;
        }
        data.game = {};
        data.game.actions = [];
        for (const row of results) {
            data.game.actions.push(JSON.parse(row.action));
        }

        done(null, socket, data);
    });
};

File: src/models/gameActions.go
package models

import (
	"database/sql"
)

type GameActions struct{}

func (*GameActions) Insert(gameID int, action string) error {
	var stmt *sql.Stmt
	if v, err := db.Prepare(`
		INSERT INTO game_actions (game_id, action)
		VALUES (?, ?)
	`); err != nil {
		return err
	} else {
		stmt = v
	}
	defer stmt.Close()

	_, err := stmt.Exec(gameID, action)
	return err
}

func (*GameActions) GetAll(databaseID int) ([]string, error) {
	var rows *sql.Rows
	if v, err := db.Query(`
		SELECT action
		FROM game_actions
		WHERE game_id = ?
		ORDER BY id
	`, databaseID); err != nil {
		return nil, err
	} else {
		rows = v
	}
	defer rows.Close()

	// Iterate over all of the actions and add them to a slice
	actions := make([]string, 0)
	for rows.Next() {
		var action string
		if err := rows.Scan(&action); err != nil {
			return nil, err
		}

		actions = append(actions, action)
	}

	return actions, nil
}
